javascript/i18n/phonenumbers/demo.js

```diff
@@ -63,57 +64,68 @@ function phoneNumberParser() {
         case PNV.TOO_LONG:
           output.append('TOO_LONG');
           break;
+        case PNV.INVALID_LENGTH:
+          output.append('INVALID_LENGTH');
+          break;
       }
       // IS_POSSIBLE shouldn't happen, since we only call this if _not_
       // possible.
       output.append('\nNote: numbers that are not possible have type ' +
           'UNKNOWN, an unknown region, and are considered invalid.');
     } else {
-      var isNumberValid = phoneUtil.isValidNumber(number);
-      output.append('\nResult from isValidNumber(): ');
-      output.append(isNumberValid);
-      if (isNumberValid && regionCode && regionCode != 'ZZ') {
+      // Checking as isValid() fails if possible local only.
+      if (isPossibleReason == PNV.IS_POSSIBLE_LOCAL_ONLY) {
+        output.append('\nResult from isPossibleNumberWithReason(): ');
+        output.append('IS_POSSIBLE_LOCAL_ONLY');
+        output.append('\nNumber is considered as invalid as it is ' +
+          'not a possible national number.');
+      } else {
+        var isNumberValid = phoneUtil.isValidNumber(number);
+        output.append('\nResult from isValidNumber(): ');
+        output.append(isNumberValid);
+        if (isNumberValid && regionCode && regionCode != 'ZZ') {
         output.append('\nResult from isValidNumberForRegion(): ');
         output.append(phoneUtil.isValidNumberForRegion(number, regionCode));
-      }
-      output.append('\nPhone Number region: ');
-      output.append(phoneUtil.getRegionCodeForNumber(number));
-      output.append('\nResult from getNumberType(): ');
-      var PNT = i18n.phonenumbers.PhoneNumberType;
-      switch (phoneUtil.getNumberType(number)) {
-        case PNT.FIXED_LINE:
-          output.append('FIXED_LINE');
-          break;
-        case PNT.MOBILE:
-          output.append('MOBILE');
-          break;
-        case PNT.FIXED_LINE_OR_MOBILE:
-          output.append('FIXED_LINE_OR_MOBILE');
-          break;
-        case PNT.TOLL_FREE:
-          output.append('TOLL_FREE');
-          break;
-        case PNT.PREMIUM_RATE:
-          output.append('PREMIUM_RATE');
-          break;
-        case PNT.SHARED_COST:
-          output.append('SHARED_COST');
-          break;
-        case PNT.VOIP:
-          output.append('VOIP');
-          break;
-        case PNT.PERSONAL_NUMBER:
-          output.append('PERSONAL_NUMBER');
-          break;
-        case PNT.PAGER:
-          output.append('PAGER');
-          break;
-        case PNT.UAN:
-          output.append('UAN');
-          break;
-        case PNT.UNKNOWN:
-          output.append('UNKNOWN');
-          break;
+        }
+        output.append('\nPhone Number region: ');
+        output.append(phoneUtil.getRegionCodeForNumber(number));
+        output.append('\nResult from getNumberType(): ');
+        var PNT = i18n.phonenumbers.PhoneNumberType;
```

Comment:
phoneNumberType

(also, can we put the switch statement below into a helper method, printType?)

Reply:
Done.